daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -1233,4 +1234,12 @@ trait ElementBase
     }
   }
 
+  private lazy val optionFloating = findPropertyOption("floating")
+
+  private def checkFloating = (optionFloating.isDefined, tunable.requireFloatingProperty) match {
+    case (false, false) => SDW(WarnID.FloatingError, "Property 'dfdl:floating' is required but not defined.")
+    case (false, true) => SDE("Property 'dfdl:floating' is required but not defined.")
```

Comment:
Might be worth replacing this SDE with a call to "floating". That call would require the floating property and output the standard property missing SDE, which also shows some context like where it looked for the property. Same with textBidi below.